functionary/core/models/package.py

```diff
@@ -73,18 +83,25 @@ def update_image_name(self, image_name: str) -> None:
     def complete(self) -> None:
         """Mark package as completed if it was previously in `PENDING` status"""
         # NOTE: This is to prevent overwriting the enabled/disabled status set by users
-        if self.status == self.PENDING:
-            self._update_status(self.COMPLETE)
+        if self.status == PENDING:
+            self._update_status(COMPLETE)
 
     def enable(self) -> None:
         """Enable package as it's associated functions"""
-        # TODO: Add method for enabling package's functions
-        self._update_status(self.ENABLED)
+        # Prevent the user from manually enabling a package that hasn't
+        # yet successfully finished being built.
+        if self.status in [COMPLETE, DISABLED]:
+            self._update_status(ENABLED)
 
     def disable(self) -> None:
         """Disable package and it's associated functions"""
-        # TODO: Add method for disabling package's functions
-        self._update_status(self.DISABLED)
+        # Prevent the user from manually disabling a package that hasn't
+        # yet successfully finished being built.
+        if self.status == ENABLED:
+            self._update_status(DISABLED)
+
+            for function in self.active_functions.all():
```

Comment:
The way I read this, you've already set the status of the package to DISABLED, so `self.active_functions.all()` will always be an empty set, won't it?  I think you want to do this whole thing in a transaction, update the scheduled tasks first, then set the package status.